stacker/actions/build.py

```diff
@@ -228,17 +233,76 @@ def _handle_missing_parameters(self, params, required_params,
 
         return params.items()
 
-    def _generate_plan(self, tail=False):
+    def _normalize_json(self, json_str):
+        """Takes a string representing a JSON object and normalizes it"""
+        obj = json.loads(json_str)
+        return json.dumps(obj, sort_keys=True, indent=4)
+
+    def _diff_stack(self, stack, **kwargs):
+        """Handles the diffing a stack in CloudFormation vs. our config"""
+        if not should_submit(stack):
+            return SKIPPED
+
+        if not should_update(stack):
+            return SKIPPED
+
+        try:
+            old_stack = self.provider.get_template(stack.fqn)
+        except exceptions.StackDoesNotExist:
+            old_stack = None
+
+        new_stack = self._normalize_json(stack.blueprint.rendered)
+
+        logger.info("============== Stack: %s ===============", stack.fqn)
+        if not old_stack:
+            logger.info("New template contents:")
+            print new_stack
+        else:
+            old_stack = self._normalize_json(old_stack)
+            old = tempfile.NamedTemporaryFile(
+                prefix="old_stack_",
+                delete=False,
+            )
+            new = tempfile.NamedTemporaryFile(
+                prefix="new_stack_",
+                delete=False,
+            )
+            try:
+                old.write(old_stack)
+                new.write(new_stack)
+                old.close()
+                new.close()
+                pobj = subprocess.Popen(
+                    ['/usr/bin/diff', DIFF_ARGS, old.name, new.name],
```

Comment:
One other thing is that this assumes that diff will be at this place, which is another reason to use difflib rather than shelling out to another command.